### ISSUE TITLE
DeprecationWarning: Please stop importing 'salt.utils.psutil_compat' and instead import 'psutil' directly as there's no longer a need for a compatability layer

### ISSUE THREAD

**Issue by TNTBOMBOM** [QubesOS/qubes-issues#9459]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release

4.2.3 (R4.2)

### Steps to reproduce

```
[user@dom0 ~]$ sudo qubesctl --show-output state.sls update.qubes-dom0
/usr/lib/python3.11/site-packages/salt/utils/psutil_compat.py:16: DeprecationWarning: Please stop importing 'salt.utils.psutil_compat' and instead import 'psutil' directly as there's no longer a need for a compatability layer. The 'salt.utils.psutil_compat' will go away on Salt 3008.0 (Argon).
  salt.utils.versions.warn_until(
local:
----------
          ID: /etc/yum.repos.d/qubes-dom0.repo
    Function: file.replace
      Result: True
     Comment: No changes needed to be made
     Started: 14:28:36.603920
    Duration: 68.045 ms
     Changes:   
----------
          ID: /etc/yum.repos.d/qubes-templates.repo
    Function: file.replace
      Result: True
     Comment: No changes needed to be made
     Started: 14:28:36.672076
    Duration: 2.058 ms
     Changes:   
----------
          ID: update
    Function: cmd.script
      Result: True
     Comment: Cache cleaned
     Started: 14:28:36.698430
    Duration: 4971.033 ms
     Changes:   
----------
          ID: update
    Function: pkg.uptodate
      Result: True
     Comment: System is already up-to-date
     Started: 14:28:44.709320
    Duration: 295479.882 ms
     Changes:   

Summary for local
------------
Succeeded: 4
Failed:    0
------------
Total states run:     4
Total run time: 300.521 s
[user@dom0 ~]$ 


```

**Comment by RhizomaticNomad** [comment]
I came across this issue because I've suddenly had the same error messages today while executing qubesctl on my user_salt states. Though it was followed by 
`[CRITICAL] Specified ext_pillar interface qvm_features is unavailable`
`[CRITICAL] Specified ext_pillar interface qvm_features is unavailable`
`[CRITICAL] Specified ext_pillar interface qvm_features is unavailable`
and then again 
`/usr/lib/python3.11/site-packages/salt/utils/psutil_compat.py:16: DeprecationWarning: Please stop importing 'salt.utils.psutil_compat' and instead import 'psutil' directly as there's no longer a need for a compatability layer. The 'salt.utils.psutil_compat' will go away on Salt 3008.0 (Argon).
  salt.utils.versions.warn_until(`
In this Issue [https://github.com/QubesOS/qubes-issues/issues/8491] someone suggested `sudo qubesctl saltutil.sync_all` and I assume this resolved the error, at least for me. (TBH, I've also run `qubesctl state.sls qubes.user-dirs` and don't know which command really fixed the issue). 
But as youwant to run the update.qubes-dom0 where no user env is involved, maybe `sudo qubesctl saltutil.sync_all` fixes your issue?

**Comment by Solomon1732** [comment]
I got the same warning when I ran the GUI update for dom0. The update itself was successful despite it. I will restart Qubes, run the `qubesctrl` command, and restart Qubes again for good measure. I hope the warning won't show in the next update. Time will tell.

I update my Qubes 4.2 regularly, so I don't know what minor version I use.

**Comment by DemiMarie** [comment]
@Solomon1732 Does `sudo qubesctl saltutil.sync_all` fix it?

**Comment by Solomon1732** [comment]
No. I wrote as much in the the edited part of my above message.

**Comment by Solomon1732** [comment]
Update: I updated Qubes again to the latest packages. This time the warning did not show up. I have no idea why.
